indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -395,6 +399,45 @@ public Response shutdown(@PathParam("id") final String id)
     return terminate(id);
   }
 
+  /**
+   * This method will immediately try to handoff the list of task group ids for the given supervisor.
+   * This is a best effort API and makes no guarantees of execution, e.g. if a non-existent task group id
+   * is passed to it, the API call will still suceced.
+   */
+  @POST
+  @Path("/{id}/taskGroups/handoff")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(SupervisorResourceFilter.class)
+  public Response handoffTaskGroups(@PathParam("id") final String id, @Nonnull final HandoffTaskGroupsRequest handoffTaskGroupsRequest)
+  {
+    List<Integer> taskGroupIds = handoffTaskGroupsRequest.getTaskGroupIds();
+    if (taskGroupIds == null || taskGroupIds.isEmpty()) {
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity(ImmutableMap.of("error", "List of task groups to handoff can't be empty"))
+          .build();
+
+    }
+    return asLeaderWithSupervisorManager(
+        manager -> {
+          try {
+            if (manager.handoffTaskGroupsEarly(id, taskGroupIds)) {
+              return Response.ok(ImmutableMap.of("id", id, "taskGroupIds", taskGroupIds)).build();
```

Comment:
In the success case, why do we return the request parameters back in the response? We might as well just return an empty 200 OK response.

Alternatively, we could return the `taskGroupIds` which were actually marked for early hand-off. In the  case where some of the requested `taskGroupIds` are non-existent or not actively reading, the returned set of `taskGroupIds` could differ from the requested one, thus telling the caller which ones will actually be handed off.